transforms/helpers/validation-helper.ts

```diff
@@ -143,36 +149,38 @@ function getLifecycleHookErrors(actionsProp: EOActionsObjectProp): string[] {
  */
 function getInfiniteLoopErrors(
   j: JSCodeshift,
-  actionsProp: EOActionsObjectProp
+  actionsProp: EOActionsProp
 ): string[] {
   const actionProps = actionsProp.properties;
   const errors: string[] = [];
   for (const actionProp of actionProps) {
     const actionName = actionProp.key.name;
-    const collection = j(actionProp.value);
-
-    // Occurrences of this.actionName()
-    const isEOActionInfiniteCall =
-      makeEOActionInfiniteCallAssertion(actionName);
-    const actionCalls = findPaths(
-      collection,
-      j.CallExpression,
-      isEOActionInfiniteCall
-    );
-
-    // Occurrences of this.get('actionName')() or get(this, 'actionName')()
-    const isEOActionInfiniteLiteral =
-      makeEOActionInfiniteLiteralAssertion(actionName);
-    const actionLiterals = findPaths(
-      collection,
-      j.Literal,
-      isEOActionInfiniteLiteral
-    );
-
-    if (actionLiterals.length > 0 || actionCalls.length > 0) {
-      errors.push(
-        `[${actionName}]: Transform not supported - calling the passed action would cause an infinite loop. See https://github.com/scalvert/eslint-plugin-ember-es6-class/pull/2 for more details`
+    if (isEOActionMethod(actionProp)) {
+      const collection = j(actionProp.body);
```

Comment:
Is this just a change from flow to babel?

Reply:
Yeah. There were pretty significant changes in flow's handling of methods vs babels.